hmy/api_backend.go

```diff
@@ -437,7 +437,43 @@ func (b *APIBackend) GetValidatorInformation(
 		return defaultReply, nil
 	}
 
-	defaultReply.Lifetime.APR = stats.APR
+	// average apr cache keys
+	key := fmt.Sprintf("apr-%s-%d", addr.Hex(), now.Uint64())
+	prevKey := fmt.Sprintf("apr-%s-%d", addr.Hex(), now.Uint64()-1)
+
+	// delete entry for previous epoch
+	b.apiCache.Forget(prevKey)
+
+	// calculate last 100 epochs for averaging APR
+	epochFrom := bc.Config().StakingEpoch
+	nowMinus100 := now.Sub(now, big.NewInt(100))
+	if nowMinus100.Cmp(epochFrom) > 0 {
+		epochFrom = nowMinus100
+	}
+
+	if avgAPR, err := b.SingleFlightRequest(
+		key, func() (interface{}, error) {
+			total := numeric.ZeroDec()
+			count := 0
+			for i := now.Int64(); i > nowMinus100.Int64(); i-- {
```

Comment:
This logic will have issue. If a validator is elected only at first and last epoch in the 100 epochs,  then the calculated APR looks normal, but actually to the delegators, the return is very little. 

better do this, start from nowMinus100, start counting (count+1) only after hitting the first present APR (to avoid new validator not having enough history), and from there, starting counting non-present APR too (the holes should be counted). finish until now.

Reply:
good point... agree.